src/components/User/Limits/components/Limits.Manager.vue

```diff
@@ -287,27 +372,56 @@ export default {
         getLimit.apply(this, [STATS_OPERATION_TYPES.withdraw]),
         getLimit.apply(this, [STATS_OPERATION_TYPES.deposit]),
       ])
-      this.limits.payment = paymentLimits
-      this.limits.withdrawal = withdrawalLimits
-      this.limits.deposit = depositLimits
-      async function getLimit (statsOpType) {
-        const requestor =
-          await this.getRequestorAccountId(this.filters.requestor)
-        const { data } = await Sdk.horizon.limits.get({
-          account_id: requestor,
-          account_type: this.filters.accountRole,
-          stats_op_type: statsOpType,
-          asset: this.filters.asset,
+
+      const limitDetails = {
+        assetCode: this.filters.asset,
+        accountId: this.filters.address,
+        accountRole: this.filters.accountRole,
+      }
+
+      this.limits.payment = new LimitsRecord(paymentLimits, {
+        statsOpType: STATS_OPERATION_TYPES.paymentOut,
+        ...limitDetails,
+      })
+
+      this.limits.withdrawal = new LimitsRecord(
+        withdrawalLimits, {
+          statsOpType: STATS_OPERATION_TYPES.withdraw,
+          ...limitDetails,
         })
-        // TODO: remove legacy consistency fix
-        if (data.accountType || data.accountType === null) {
-          const role = data.accountType === null
-            ? null
-            : String(data.accountType)
-          data.accountRole = role
-          delete data.accountType
+
+      this.limits.deposit = new LimitsRecord(
+        depositLimits, {
+          statsOpType: STATS_OPERATION_TYPES.deposit,
+          ...limitDetails,
+        })
+
+      async function getLimit (statsOpType) {
+        const filters = {}
+        if (this.filters.scope === SCOPE_TYPES.account) {
+          filters.account = this.filters.address
+        } else if (this.filters.scope === SCOPE_TYPES.accountRole) {
+          filters.account_role = this.filters.accountRole
+        }
+        const { data: limits } = await ApiCallerFactory
+          .createCallerInstance()
+          .getWithSignature('/v3/limits', {
+            filter: {
+              asset: this.filters.asset,
+              stats_op_type: statsOpType,
+              ...filters,
+            },
+          })
+        if (this.filters.scope === SCOPE_TYPES.global) {
+          const globalLimits = limits.filter((limit) => {
+            if (!limit.accountRole && !limit.account) {
+              return limit
+            }
```

Comment:
filter’s callback expects boolean to be returned, so you can return like
```js
const globalLimits = limits.filter((limit) => {
  return !limit.accountRole && !limit.account
}
```
or
```js
const globalLimits = limits.filter((limit) => {
  return !(limit.accountRole || limit.account)
}
```
or 
```js
const globalLimits = limits
  .filter((limit) => !(limit.accountRole || limit.account))
```